core/domain/state_domain.py

```diff
@@ -95,13 +95,41 @@ def from_dict(cls, answer_group_dict):
         Returns:
             AnswerGroup. The corresponding AnswerGroup domain object.
         """
+        if not isinstance(answer_group_dict['rule_specs'], list):
+            raise Exception(
+                'Expected answer group rule specs to be a list, '
+                'received %s' % answer_group_dict['rule_specs'])
+
         return cls(
             Outcome.from_dict(answer_group_dict['outcome']),
             [RuleSpec.from_dict(rs) for rs in answer_group_dict['rule_specs']],
             answer_group_dict['training_data'],
             answer_group_dict['tagged_skill_misconception_id']
         )
 
+    @classmethod
+    def from_dict_list(cls, answer_group_dict_list):
+        """Return a AnswerGroup list domain objects from a dict list.
```

Comment:
Creates a list of AnswerGroup dict into a list of objects.

Reply:
done